lib/matplotlib/axes/_base.py

```diff
@@ -105,8 +98,8 @@ def _process_plot_format(fmt):
                 raise ValueError(
                     'Illegal format string "%s"; two color symbols' % fmt)
             color = c
-        elif c == 'C' and i < len(chars) - 1:
-            color_cycle_number = int(chars[i + 1])
+        elif c == 'C' and i < len(fmt) - 1:
+            color_cycle_number = int(fmt[i + 1])
             color = mcolors.to_rgba("C{}".format(color_cycle_number))
             i += 1
```

Comment:
I would inline the increment completely into every if block. Having a global +1 and sometimes and additional +1 is more confusing than +1 or +2 in every block.